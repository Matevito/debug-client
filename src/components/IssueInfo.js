import React from 'react'

export const IssueInfo = ({ issue }) => {
    return (
        <div>
            issuepapper
        </div>
    )
}
