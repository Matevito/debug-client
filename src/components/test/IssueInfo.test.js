import React from "react";

// testing components
import { render, screen } from "@testing-library/react";
import '@testing-library/jest-dom'

// test comp
import { IssueInfo } from "../IssueInfo";

describe("IssueInfo component", () => {
    let issueInfo; 
    let userId;

    beforeEach(() => {
        // set-up vars
        userId = "testUserId"
        issueInfo = {
            _id: "testIssue",
            title: "testing title",
            description: "a simple description of elevating a div",
            date : "2022-04-23T22:56:09.627Z",
            handlingTeam: [],
            priority: "low",
            project: "testProject",
            screenshots: [],
            status: "aditional info needed",
            type: "bugg-error"
        };
    })
    test("renders issue information", () => {
        render(
            <IssueInfo 
                issue={issueInfo}
                userId={userId}
            />
        )
        expect(screen.getByText("Ticket information")).toBeInTheDocument()
        expect(screen.getByText("testing title")).toBeInTheDocument()
        expect(screen.getByText("a simple description of elevating a div")).toBeInTheDocument()
        expect(screen.getByText("low")).toBeInTheDocument()
        expect(screen.getByText("aditional info needed")).toBeInTheDocument()
        expect(screen.getByText("bugg-error")).toBeInTheDocument()
        expect(screen.getByText("Apr 23, 2022")).toBeInTheDocument()
        expect(screen.getByText("Take ticket!")).toBeInTheDocument()
        expect(screen.getByText("No screenshots")).toBeInTheDocument();
    });
    test("handles screenshots conditional", () => {
        issueInfo.screenshots = ["url1", "url2"]; 
        render(
            <IssueInfo 
                issue={issueInfo}
                userId={userId}
            />
        )
        expect(screen.getByText("Screenshots")).toBeInTheDocument();
        expect(screen.getByText("Screenshot 1")).toBeInTheDocument();
        expect(screen.getByText("Screenshot 2")).toBeInTheDocument();
    });
    test("handles issue without team", () => {
        render(
            <IssueInfo 
                issue={issueInfo}
                userId={userId}
            />
        )
        expect(screen.getByText("No users assigned yet!")).toBeInTheDocument();
    });
    test("handles user that has taken the ticket", () => {
        issueInfo.handlingTeam = [{
            _id: userId, username: "testUser1"
        }];
        render(
            <IssueInfo 
                issue={issueInfo}
                userId={userId}
            />
        )
        expect(screen.getByText("Team")).toBeInTheDocument();
        expect(screen.getByText("testUser1")).toBeInTheDocument();
        expect(screen.getByText("Leave ticket!")).toBeInTheDocument();
    });
    test("check snapshot", () => {
        const { container } = render(<IssueInfo issue={issueInfo} userId={userId}/>)
        expect(container).toMatchSnapshot();
    });
})