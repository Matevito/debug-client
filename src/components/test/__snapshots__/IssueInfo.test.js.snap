// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`IssueInfo component check snapshot 1`] = `
<div>
  <div
    class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation8 MuiCard-root css-fky5tw-MuiPaper-root-MuiCard-root"
    xs="[object Object]"
  >
    <div
      class="MuiCardContent-root css-2of5mt-MuiCardContent-root"
    >
      <h6
        class="MuiTypography-root MuiTypography-h6 css-2ulfj5-MuiTypography-root"
      >
        Ticket information
      </h6>
      <button
        class="MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium MuiButtonBase-root  css-1e6y48t-MuiButtonBase-root-MuiButton-root"
        tabindex="0"
        type="button"
      >
        Take ticket!
        <span
          class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
        />
      </button>
      <div
        class="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-2 css-mhc70k-MuiGrid-root"
      >
        <div
          class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-6 css-1hn59hz-MuiGrid-root"
        >
          <span
            class="MuiTypography-root MuiTypography-h7 css-37jdci-MuiTypography-root"
          >
            <b>
              Title
            </b>
          </span>
          <p
            class="MuiTypography-root MuiTypography-body1 css-ahj2mt-MuiTypography-root"
          >
            testing title
          </p>
          <span
            class="MuiTypography-root MuiTypography-h7 css-37jdci-MuiTypography-root"
          >
            <b>
              Description
            </b>
          </span>
          <p
            class="MuiTypography-root MuiTypography-body1 css-ahj2mt-MuiTypography-root"
          >
            a simple description of elevating a div
          </p>
          <span
            class="MuiTypography-root MuiTypography-h7 css-37jdci-MuiTypography-root"
          >
            <b>
              Created
            </b>
          </span>
          <p
            class="MuiTypography-root MuiTypography-body1 css-ahj2mt-MuiTypography-root"
          >
            Apr 23, 2022
          </p>
          <span
            class="MuiTypography-root MuiTypography-h7 css-37jdci-MuiTypography-root"
          >
            <b>
              No screenshots
            </b>
          </span>
          <div
            class="MuiBox-root css-1xhj18k"
          />
        </div>
        <div
          class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-6 css-1hn59hz-MuiGrid-root"
        >
          <span
            class="MuiTypography-root MuiTypography-h7 css-37jdci-MuiTypography-root"
          >
            <b>
              Priority
            </b>
          </span>
          <p
            class="MuiTypography-root MuiTypography-body1 css-ahj2mt-MuiTypography-root"
          >
            low
          </p>
          <span
            class="MuiTypography-root MuiTypography-h7 css-37jdci-MuiTypography-root"
          >
            <b>
              Type
            </b>
          </span>
          <p
            class="MuiTypography-root MuiTypography-body1 css-ahj2mt-MuiTypography-root"
          >
            bugg-error
          </p>
          <span
            class="MuiTypography-root MuiTypography-h7 css-37jdci-MuiTypography-root"
          >
            <b>
              Status
            </b>
          </span>
          <p
            class="MuiTypography-root MuiTypography-body1 css-ahj2mt-MuiTypography-root"
          >
            aditional info needed
          </p>
          <span
            class="MuiTypography-root MuiTypography-h7 css-37jdci-MuiTypography-root"
          >
            <b>
              No users assigned yet!
            </b>
          </span>
        </div>
      </div>
    </div>
  </div>
</div>
`;
